### PR TITLE
Image preview on a map

### PR DESCRIPTION
**Pull request by odinkomnogim**
New version of 'image_preview_l' branch without conflicts

### CODE REVIEW

File: mapflow/functional/api/data_catalog_api.py [hunk 1]
```diff
@@ -157,6 +165,51 @@ def get_image_preview(self,
...
+                            callback: Callable,
+                            image_id: str = ""):
+        self.http.get(url=image.preview_url_l,
+                      callback=callback,
+                      use_default_error_handler=False,
+                      error_handler=self.image_preview_l_error_handler,
+                      callback_kwargs={"extent": extent,
+                                       "image_id": image_id})
+
+    def display_image_preview(self,
```
Comment (AlexeyTrekin) [inline mapflow/functional/api/data_catalog_api.py:180]: I don't remember, maybe we discussed it to be like this, but I think that display_image_preview contains some app logic, so it belongs to service, not API

File: mapflow/functional/layer_utils.py [hunk 2]
```diff
@@ -229,6 +230,14 @@ def generate_vector_layer(layer_uri,
     )
     return layer
 
+def footprint_to_extent(footprint):
```
Comment (AlexeyTrekin) [inline mapflow/functional/layer_utils.py:233]: Please, use type annotations and return type hints. 
It is especially important in the functional module as it is used outside of the module and we need to keep track without documentation.

Also the docstrings may be of some use here, but in many cases types + function name are enough to understand what it is doing.


```def footprint_to_extent(footprint: dict) -> QgsRectangle:```
(check the types I used here! I did not check them)

Also, as th type of footprint is just a plain dict and it does not say anything particular about the input data expected, you may do one of two:
- (easy) add docstrings, specify that it is a `geojson-like dict (geometry)`
- best, but let's skip it by now: add a special class to represent the geojson-like geometry and use this class here (and in ImageReturnSchema as well)

File: mapflow/functional/layer_utils.py [hunk 3]
```diff
@@ -610,4 +619,4 @@ def set_raster_extent_error_handler(self,
         """
         self.add_layer(vector)
         self.iface.setActiveLayer(vector)
-        self.iface.zoomToActiveLayer()
+        self.iface.zoomToActiveLayer()
```
Comment (AlexeyTrekin) [inline mapflow/functional/layer_utils.py:622]: Why did you remove the empty line at the end? The blank line is strongly recommended (it is default file ending in UNIX systems)

**Comment by odinkomnogim** [comment]
New version of this branch without conflicts ('image_preview_large') was merged into 'feature/data-catalog-api'
